### ISSUE TITLE
Missing permission in ArgoRollout cluster role

### ISSUE THREAD

**Issue by Ahmed-Elkollaly** [argoproj/argo-helm#2795]
### Describe the bug

I can't use the recent feature `scaleDown`, used in migration to Argo rollout, due to missing permission in `clusterrole` in the helm chart in argo-rollout v2.36.1 https://github.com/argoproj/argo-helm/blob/main/charts/argo-rollouts/templates/controller/clusterrole.yaml#L60

https://argo-rollouts.readthedocs.io/en/stable/migrating/
```
  workloadRef:                                 # Reference an existing Deployment using workloadRef field
    apiVersion: apps/v1
    kind: Deployment
    name: rollout-ref-deployment
    scaleDown: onsuccess
```
The error from argo-rollout controller pod
```time="2024-06-25T12:38:07Z" level=error msg="deployments.apps \"rollout-ref-deployment\" is forbidden: User \"system:serviceaccount:argo-rollouts:argo-rollouts\" cannot update resource \"deployments\" in API group \"apps\" in the namespace \"dev\"```

### Related helm chart

argo-rollouts

### Helm chart version

2.36.1

### To Reproduce

Steps to reproduce the issue
1. Apply deployment
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app.kubernetes.io/instance: rollout-canary
  name: rollout-ref-deployment
spec:
  replicas: 1                              
  selector:
    matchLabels:
      app: rollout-ref-deployment
  template:
    metadata:
      labels:
        app: rollout-ref-deployment
    spec:
      containers:
        - name: rollouts-demo
          image: argoproj/rollouts-demo:blue
          imagePullPolicy: Always
          ports:
            - containerPort: 8080
```
            
 2. Apply Rollout
 
```
apiVersion: argoproj.io/v1alpha1               # Create a rollout resource
kind: Rollout
metadata:
  name: rollout-ref-deployment
spec:
  replicas: 5
  selector:
    matchLabels:
      app: rollout-ref-deployment
  workloadRef:                                 # Reference an existing Deployment using workloadRef field
    apiVersion: apps/v1
    kind: Deployment
    name: rollout-ref-deployment
    scaleDown: onsuccess
  strategy:
    canary:
      steps:
        - setWeight: 20
        - pause: {duration: 10s}
```

3. check argo-rollout controller pod logs
`time="2024-06-25T12:38:07Z" level=error msg="deployments.apps \"rollout-ref-deployment\" is forbidden: User \"system:serviceaccount:argo-rollouts:argo-rollouts\" cannot update resource \"deployments\" in API group \"apps\" in the namespace \"dev\"`

### Expected behavior

use the recent feature `scaleDown` in argorollout v1.7.0 without any access issue

**Comment by yu-croco** [comment]
Hi @Ahmed-Elkollaly , thank you for opening issue.
Since we follow upstream's manifest (in this case, [here](https://github.com/argoproj/argo-rollouts/blob/master/manifests/role/argo-rollouts-clusterrole.yaml#L59-L69)), can you also open an issue in upstream?
Once upstream releases new version, we also follow it.

**Comment by Ahmed-Elkollaly** [comment]
I opened an [issue](https://github.com/argoproj/argo-rollouts/issues/3672) in the argo-rollout project. thank you

**Comment by yu-croco** [comment]
📝  
upstream has been fixed in https://github.com/argoproj/argo-rollouts/pull/3675 . We will follow the fix when the PR is delivered as a specific version.

**Comment by DrFaust92** [comment]
Ill follow up on this and open a PR here now thats its merged upstream

**Comment by yu-croco** [comment]
Hi @DrFaust92 
FYI; Renovate in argo-helm will open PR when upstream releases new version, and we will check the diffs under manifest in upstream at the PR. :)
e.g.) https://github.com/argoproj/argo-helm/pull/2794

**Comment by DrFaust92** [comment]
[yu-croco](https://github.com/yu-croco) makes sense, but in this case it was a manifest only upstream change and doesnt require a binary release to add this.

**Comment by yu-croco** [comment]
We only follow specific version of upstream, instead of main branch.
Upstream will release the manifest diff with other diffs, so we need to wait for next (or next next?) release of upstream.

**Comment by muma378** [comment]
Hi I had the same issue when use the `scaledown` feature. I found the feature was released in `v1.7.1`  and the image was updated in helm, so is it the right time to update the cluster role? 
I opened a PR #2816, please feel free to close if it is not correct.

**Comment by yu-croco** [comment]
Hi @muma378 , 
As I mentioned [earlier](https://github.com/argoproj/argo-helm/issues/2795#issuecomment-2190502652), the point is upstresm's manifest.
There is no `update` permission in v1.7.1 (ref: [v1.7.1's permission](https://github.com/argoproj/argo-rollouts/blob/v1.7.1/manifests/role/argo-rollouts-clusterrole.yaml#L59-L69)), so we are waiting for it.

**Comment by muma378** [comment]
> Hi @muma378 , As I mentioned [earlier](https://github.com/argoproj/argo-helm/issues/2795#issuecomment-2190502652), the point is upstresm's manifest. There is no `update` permission in v1.7.1 (ref: [v1.7.1's permission](https://github.com/argoproj/argo-rollouts/blob/v1.7.1/manifests/role/argo-rollouts-clusterrole.yaml#L59-L69)), so we are waiting for it.

OK I got it, wo we are waiting for `v1.7.2` to release. I thought it is the time to update cause I found the upstream's manifest was fixed in https://github.com/argoproj/argo-helm/pull/2816.

**Comment by amorattab** [comment]
@Ahmed-Elkollaly addtional RBAC rules can be added to the cluster role through providerRBAC.additionalRules. 
Setting the following in the values.yaml file fixed the issue:

providerRBAC:
    additionalRules:
        - apiGroups: 
           - apps
        - resources:
           - deployments
        - verbs:
           - update

**Comment by yu-croco** [comment]
This issue is fixed in argo-rollouts v1.7.2 👍 
https://github.com/argoproj/argo-helm/pull/2880
